lemmy_utils/src/email.rs

```diff
@@ -18,35 +20,61 @@ pub fn send_email(
   html: &str,
 ) -> Result<(), String> {
   let email_config = Settings::get().email.ok_or("no_email_setup")?;
+  let domain = &Settings::get().hostname;
 
-  let email = Email::builder()
-    .to((to_email, to_username))
-    .from(email_config.smtp_from_address.to_owned())
+  let email = Message::builder()
+    .from(
+      email_config
+        .smtp_from_address
+        .parse()
+        .expect("from address isn't valid"),
+    )
+    .to(Mailbox::new(
+      Some(to_username.to_string()),
+      Address::from_str(to_email).expect("to address isn't valid"),
+    ))
     .subject(subject)
-    .html(html)
-    .build()
-    .unwrap();
-
-  let mailer = if email_config.use_tls {
-    SmtpClient::new_simple(&email_config.smtp_server).unwrap()
-  } else {
-    SmtpClient::new(&email_config.smtp_server, ClientSecurity::None).unwrap()
+    .multipart(
+      MultiPart::mixed().multipart(
+        MultiPart::alternative()
+          .singlepart(
+            SinglePart::eight_bit()
+              .header(header::ContentType(
+                "text/plain; charset=utf8".parse().unwrap(),
+              ))
+              .body(html),
+          )
+          .multipart(
+            MultiPart::related().singlepart(
+              SinglePart::eight_bit()
+                .header(header::ContentType(
+                  "text/html; charset=utf8".parse().unwrap(),
+                ))
+                .body(html),
+            ),
+          ),
+      ),
+    )
+    .expect("email built correctly");
+
+  let mut builder = SmtpTransport::builder_dangerous(domain);
```

Comment:
This is because the SMTP runs in Docker and doesnt have TLS? I would add a comment to explain this.

Reply:
I'm not sure tbh, @paolobarbolini left this note for that:

 // don't worry about 'dangeous'. it's just that leaving it at the default configuration is bad.

Reply:
Yup, TLS gets configured a few lines later. It's called `builder_dangerous` because the default configuration doesn't encrypt the connection and uses port 25

Reply:
Still, I would leave a comment here because this looks pretty bad if someone comes across it by chance.

Reply:
I can do.